src/config.rs

```diff
@@ -784,14 +786,18 @@ impl<'a> Cfg<'a> {
         components: Vec<String>,
         targets: Vec<String>,
         profile: Option<Profile>,
-    ) -> Result<Toolchain<'_>> {
+        verbose: bool,
+    ) -> Result<(UpdateStatus, Toolchain<'_>)> {
+        if verbose {
```

Comment:
The notifications aren't mentioned in the commit message, and don't seem to be related to the return type. Separate commit?

Reply:
@djc This is very weird, could be a rebase error on my side.

Reply:
@djc Oh, it looks like `git absorb` has messed up my history :( I'll be more careful next time.

Reply:
The original commit was there: https://github.com/rust-lang/rustup/commit/07777feac5e4b85846662335c541a2a57222238e

Reply:
I cherry-picked that one out. The history should be okay now.